paasta_tools/spark_tools.py

```diff
@@ -169,13 +170,27 @@ def get_volumes_from_spark_mesos_configs(spark_conf: Mapping[str, str]) -> List[
 
 
 def get_volumes_from_spark_k8s_configs(spark_conf: Mapping[str, str]) -> List[str]:
-    volume_names = [
-        re.match(
-            r"spark.kubernetes.executor.volumes.hostPath.(\d+).mount.path", key
-        ).group(1)
-        for key in spark_conf.keys()
-        if "spark.kubernetes.executor.volumes.hostPath." in key and ".mount.path" in key
-    ]
+    volume_names = []
+    for key in list(spark_conf.keys()):
+        if (
+            "spark.kubernetes.executor.volumes.hostPath." in key
+            and ".mount.path" in key
+        ):
+            v_name = re.match(
+                r"spark.kubernetes.executor.volumes.hostPath.([a-z0-9]([-a-z0-9]*[a-z0-9])?).mount.path",
+                key,
+            )
+            if v_name:
+                volume_names.append(v_name.group(1))
+            else:
+                log.error(
+                    f"Volume names must consist of lower case alphanumeric characters or '-', "
+                    f"and must start and end with an alphanumeric character. Config -> '{key}' must be fixed."
+                )
+                # Failing here because the k8s pod fails to start if the volume names
+                # don't follow the lowercase RFC 1123 standard.
+                sys.exit(1)
```

Comment:
What if this config is just removed from spark_conf instead of existing?

Reply:
Can you add a unit test-case for the warning/failure and success scenario?

Reply:
I'll have to remove the corresponding configs like `spark.kubernetes.executor.volumes.hostPath.{volume_name}.options.path` and `spark.kubernetes.executor.volumes.hostPath.{volume_name}.mount.readOnly` as well or it'll fail on its own. I thought it's better to let the user fix it instead of removing

Reply:
Sure I'll add a unit test